FinSearchUnified/Tests/Components/ProductStream/RepositoryTest.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+namespace FinSearchUnified\Tests\Components\ProductStream;
+
+use Enlight_Controller_Request_RequestHttp as RequestHttp;
+use FinSearchUnified\Components\ProductStream\CriteriaFactory;
+use FinSearchUnified\Components\ProductStream\Repository;
+use FinSearchUnified\Constants;
+use Shopware\Bundle\SearchBundle\Condition\CategoryCondition;
+use Shopware\Bundle\SearchBundle\Criteria;
+use Shopware\Bundle\StoreFrontBundle\Service\ContextServiceInterface;
+use Shopware\Components\Test\Plugin\TestCase;
+
+class RepositoryTest extends TestCase
+{
+    /**
+     * Provider for test cases to test CriteriaFactory
+     *
+     * @return array
+     */
+    public function shopSearchSwitchProvider()
+    {
+        return [
+            'Uses the original implementation' => [
+                'ActivateFindologic' => true,
+                'ShopKey' => 'ABCD0815',
+                'ActivateFindologicForCategoryPages' => false,
+                'findologicDI' => false,
+                'isSearchPage' => false,
+                'isCategoryPage' => true,
+                'prepareCriteria' => true,
+                'expected' => Shopware()->Shop()->getCategory()->getId()
+            ],
+            'Uses the original implementation for backend' => [
+                'ActivateFindologic' => true,
+                'ShopKey' => 'ABCD0815',
+                'ActivateFindologicForCategoryPages' => false,
+                'findologicDI' => false,
+                'isSearchPage' => true,
+                'isCategoryPage' => false,
+                'prepareCriteria' => false,
+                'expected' => Shopware()->Shop()->getCategory()->getId()
+            ],
+            'Uses the custom implementation' => [
+                'ActivateFindologic' => true,
+                'ShopKey' => 'ABCD0815',
+                'ActivateFindologicForCategoryPages' => false,
+                'findologicDI' => false,
+                'isSearchPage' => true,
+                'isCategoryPage' => false,
+                'prepareCriteria' => false,
+                'expected' => -1
+            ]
+        ];
+    }
+
+    /**
+     * @dataProvider shopSearchSwitchProvider
+     *
+     * @param bool $isActive
+     * @param string $shopKey
+     * @param bool $isActiveForCategory
+     * @param bool $checkIntegration
+     * @param bool $isSearchPage
+     * @param bool $isCategoryPage
+     * @param bool $prepareCriteria
+     * @param int $expected
+     *
+     * @throws \Enlight_Exception
+     */
+    public function testCreateCriteria(
```

Comment:
Something looks odd here. It seems that you are testing `ProductStream\CriteriaFactory` before `ProductStream\Repository` is tested. That shouldn't be necessary since there is already a separate test class for `ProductStream\CriteriaFactory`. So:
* Remove everything that is not related to `ProductStream\Repository`.

Reply:
But it was mentioned in the ticket to implement test cases as for CriteriaFactory. What do you mean by that?